### DISCUSSION TITLE
How to use the notification URL?

### DISCUSSION THREAD

**Discussion by jdswinbank** [swiftbar/SwiftBar#122] [Q&A]
I'm excited to use the notification URL added in the recent release!

But... I can't figure out how. Hitting the URL doesn't seem to do anything, and I noticed that SwiftBar doesn't appear in the list of apps allowed to produce notifications in the System Preferences.

Am I missing the obvious?

**Comment by melonamin** [comment]
>Am I missing the obvious?

No, you are not missing anything, this is on me - I'm a dummy and pushed this without proper permission handling which was OK during development testing but is broken in release...

Can you please try this build and if it works I'll update the public one: [SwiftBar.zip](https://github.com/swiftbar/SwiftBar/files/5758045/SwiftBar.zip)

**Reply by jdswinbank** [reply]
Thank you for the swift reply, and for the reassurance that I'm not as dim as I thought. However, I tried that build and didn't notice any obvious difference in behaviour — I still can't persuade it to generate any notifications, and nor did it show up in System Preferences.

Don't know if it's helpful, but I do see messages like the following in Console.app:

```
Acquiring assertion targeting [app<application.com.ameba.SwiftBar.59119158.59119170(501)>:9218] from originator [daemon<com.apple.coreservices.launchservicesd>:99] with description <RBSAssertionDescriptor| "notification:9218" ID:170-99-2298 target:9218 attributes:[
	<RBSDomainAttribute| domain:"com.apple.launchservicesd" name:"LSNotification" sourceEnvironment:"(null)">
	]>
```

**Reply by melonamin** [reply]
Are you on Catalina or Big Sur?

**Reply by jdswinbank** [reply]
Big Sur.

**Reply by melonamin** [reply]
Ok, I've tested this on a clean Big Sur install - it should ask for permission to show notifications and appear in system preferences pane.

I'm not sure, but I think you have to delete all other instances of SwiftBar, otherwise URL scheme will not be working.

New build  [SwiftBar.zip](https://github.com/swiftbar/SwiftBar/files/5761203/SwiftBar.app.zip)

**Reply by bilal-fazlani** [reply]
Notifications are working for me on Catalina btw.

**Reply by jdswinbank** [reply]
Thanks!

Unfortunately: I deleted all instances of SwiftBar, rebooted (hey, you never know), downloaded this build, started it up, and... still nothing. Perhaps this is a case of PEBKAC. 🤷

**Reply by melonamin** [reply]
Well, it's time for the big guns... can you share a screen recording of you trying to use this feature?

GitHub supports .mov files, so you can use QuickTime to record your screen.

**Reply by jdswinbank** [reply]
I made a beautiful movie — seriously, Kubrick would be quaking in his boots — but GitHub tells me its too large to upload. I think this screenshot shows the salient points, though. You'll have to take my word for it that no notifications are being displayed.

<img width="1357" alt="Screenshot 2021-01-04 at 22 08 47" src="https://user-images.githubusercontent.com/460854/103579962-9d5e3a00-4ed9-11eb-9d68-1949a4dfbb56.png">

**Reply by melonamin** [reply]
Hm, maybe I know what's going on - plugin name is the full filename, so in your example the URL should look like this:

```
swiftbar://notify?plugin=MyPlugin.1m.py&title=title&subtitle=subtitle&body=body&silent=true
```

I hope this works 🤞

**Reply by jdswinbank** [reply]
Got it! That does indeed work; thank you. 👍
